extract.py

```diff
@@ -0,0 +1,81 @@
+"""Script that connects to 'data-eng-plants-api'
+   Extracts the data
+   Puts it into a Pandas DataFrame"""
+import logging
+from multiprocessing import Pool, cpu_count
+import requests
+import pandas as pd
+
+
+URL = "https://data-eng-plants-api.herokuapp.com/plants/"
+TOTAL_NUMBER_OF_PLANTS = 50
+
+
+def get_plant_data(plant_id: int) -> dict:
+    """GET request to collect plant data from API."""
+    try:
+        response = requests.get(URL + str(plant_id), timeout=1000)
+        if response.status_code == 200:
+            return response.json()
+        return {"error": 404, "message": f"Failed to retrieve data for plant ID {plant_id}."}
+    except requests.RequestException as e:
+        logging.error("Request failed for plant ID %d: %s", plant_id, e)
+        return {"error": 500, "message": f"Server error for plant ID {plant_id}."}
```

Comment:
Likewise for this, does the code reaching this except ALWAYS indicate a 500 error?